### PR TITLE
[AdminBundle] Refactor Document controllers to share more code

### LINKED ISSUE #8843: Refactor Document controllers to share more code & make it less complex

### PR DESCRIPTION
**Pull request by dvesh3**
## Changes in this pull request  
Resolves #8843

### CODE REVIEW

**Review by dvesh3 — COMMENTED** [review]
we should check if Pimcore bundles are extending Document Controller and fix the routing, if it is the case. Like headless documents https://github.com/pimcore/headless-documents/blob/1.x/src/Controller/Admin/Document/HeadlessDocumentController.php#L38

**Comment by mcop1** [comment]
There seems to be a problem with autosave functionality

1.) Open an existing page
2.) Change the title (or change anything really)
3.) Wait for autosave 
4.) Error Msg:
image
5.) Clicking the save button works

**Comment by dvesh3** [comment]
> There seems to be a problem with autosave functionality
> 
> 1.) Open an existing page 2.) Change the title (or change anything really) 3.) Wait for autosave 4.) Error Msg: image 5.) Clicking the save button works

@mcop1 Fixed with correct task matching & permissions check

File: bundles/AdminBundle/Controller/Admin/Document/DocumentControllerBase.php [hunk 1]
```diff
@@ -427,9 +427,10 @@ protected function saveDocument(Model\Document $document, Request $request, bool
                 $document->save();
 
                 break;
-            case $task === self::TASK_SAVE && $document->isAllowed($task):
-            case $task === self::TASK_VERSION && $document->isAllowed($task):
-            case $task === self::TASK_AUTOSAVE && $document->isAllowed($task):
+            case ($task === self::TASK_SAVE
+                || $task === self::TASK_VERSION
+                || $task === self::TASK_AUTOSAVE)
+                && $document->isAllowed(self::TASK_SAVE):
```
Comment (jdreesen) [inline bundles/AdminBundle/Controller/Admin/Document/DocumentControllerBase.php]: What about: 
```suggestion
            case in_array($task, [self::TASK_SAVE, self::TASK_VERSION, self::TASK_AUTOSAVE]) 
                && $document->isAllowed(self::TASK_SAVE):
```

**Review by mcop1 — APPROVED** [review]

**Pull request merged**